### PR TITLE
I2C write register:   multi byte register may fail with large data

### PR DESCRIPTION
**Pull request by taartspi**
One bug fix.  In the last review of two byte I2C register read/write, I made a copy paste error effecting writes of large number of data bytes. 

Also, added testcases for these new two byte register I2C reads and writes.  Open to any suggestions on how I added usage of their interfaces I2cRegisterRead and I2cRegisterWriter

### CODE REVIEW

File: plugins/pi4j-plugin-mock/src/main/java/com/pi4j/plugin/mock/provider/i2c/MockI2C.java [hunk 1]
```diff
@@ -305,9 +325,36 @@ public int readRegister(int register) {
...
+        logger.info(" [");
+        logger.info(Mock.I2C_PROVIDER_NAME);
+        logger.info("::");
+        logger.info(this.id);
+        logger.info("] :: READREGISTER(");
+        logger.info("REG= (two byte offset LSB first)");
+        logger.info(StringUtil.toHexString(register, 0, register.length));
+        logger.info("offset = " + String.format("%02X", offset));
+        logger.info("Chip register offset Decimal : " + internalOffset + "  Hex : " +  String.format("%02X", internalOffset));
+        logger.info(String.valueOf(register));
```
Comment (eitch) [inline plugins/pi4j-plugin-mock/src/main/java/com/pi4j/plugin/mock/provider/i2c/MockI2C.java:353]: qodana says calling implicit toString() of register won't work

File: pi4j-test/src/test/java/com/pi4j/test/io/i2c/I2CRegisterDataTest.java [hunk 2]
```diff
@@ -117,23 +135,220 @@ public void testRegisterDataWriteRead() {
...
+     *  Unit testing, was used for unit testing write to register 0x0142 and read from 0x0143.
+     *  Not supported using ArrayDeque, required flat multi dimension array
+     *
+     * @param a         byte array, comparison begins at aOffset
+     * @param b         byte array whose contents are compared for numBytes
+     * @param aOffset   starting compare offset within 'a'
+     * @param numBytes  How many bytes to compare
+     * @return          If comparison equal, true, else false
+     */
+    public boolean compArray( byte[] a, byte[]b, int aOffset, int numBytes){
```
Comment (eitch) [inline pi4j-test/src/test/java/com/pi4j/test/io/i2c/I2CRegisterDataTest.java:336]: The name of this method should be compareArray()

File: pi4j-test/src/test/java/com/pi4j/test/io/i2c/I2CRegisterDataTest.java [hunk 3]
```diff
@@ -117,23 +135,220 @@ public void testRegisterDataWriteRead() {
...
+            case 7:
+                break;
+            case 8:
+                break;
+            case 9:
+                break;
+            case 10:
+                break;
+            default:
+                ;//Assertions.fail("writeRegisterData: invalid switchNumber");
```
Comment (eitch) [inline pi4j-test/src/test/java/com/pi4j/test/io/i2c/I2CRegisterDataTest.java:321]: Please remove the commented out code

File: pi4j-test/src/test/java/com/pi4j/test/io/i2c/I2CRegisterDataTest.java [hunk 4]
```diff
@@ -117,23 +135,220 @@ public void testRegisterDataWriteRead() {
...
+                // read back register contents from offset 0x0142. Compare offset 0x05 in bigArray
+                readData = new byte[0x10];
+                tbReg[0] = 0x42;
+                tbReg[1] = 0x01;
+                rc = register.readRegister(tbReg, readData, 0, 0x10);
+                if(rc != 0x10){
+                    Assertions.fail("readRegisterData: fail  RC switchNumber : "  + sample.switchNumber);
+                }
+                compare = this.compArray(sample.bigArray, readData, 0x05, 0x10);
+                if(compare == false){
```
Comment (eitch) [inline pi4j-test/src/test/java/com/pi4j/test/io/i2c/I2CRegisterDataTest.java:236]: !compare

File: pi4j-test/src/test/java/com/pi4j/test/io/i2c/I2CRegisterDataTest.java [hunk 5]
```diff
@@ -117,23 +135,220 @@ public void testRegisterDataWriteRead() {
...
+                // do the read back verification
+                // read back register contents from offset 0x42. Compare offset 0x05 in bigArray
+                readData = new byte[0x10];
+                obReg = 0x42;
+                rc = register.readRegister(obReg, readData, 0, 0x10);
+                if(rc != 0x10){
+                    Assertions.fail("readRegisterData: fail  RC switchNumber : "  + sample.switchNumber);
+                }
+                compare = this.compArray(sample.bigArray, readData, 0x5, 0x10);
+                if(compare == false){
```
Comment (eitch) [inline pi4j-test/src/test/java/com/pi4j/test/io/i2c/I2CRegisterDataTest.java:221]: !compare

File: pi4j-test/src/test/java/com/pi4j/test/io/i2c/I2CRegisterDataTest.java [hunk 6]
```diff
@@ -117,23 +135,220 @@ public void testRegisterDataWriteRead() {
...
+                // read back register contents from offset 0x0142. Compare offset 0x0 in bigArray
+                readData = new byte[0x10];
+                tbReg[0] = 0x42;
+                tbReg[1] = 0x01;
+                rc = register.readRegister(tbReg, readData, 0, 0x10);
+                if(rc != 0x10){
+                    Assertions.fail("readRegisterData: fail  RC switchNumber : "  + sample.switchNumber);
+                }
+                compare = this.compArray(sample.bigArray, readData, 0x0, 0x10);
+                if(compare == false){
```
Comment (eitch) [inline pi4j-test/src/test/java/com/pi4j/test/io/i2c/I2CRegisterDataTest.java:193]: this should be a !compare

**Review by eitch — COMMENTED** [review]
Please check my comments, otherwise thanks for this MR. Looks good!

File: README.md [hunk 7]
```diff
@@ -5,6 +5,7 @@
 Build state: 
 ![GitHub Actions build state](https://github.com/pi4j/pi4j-v2/workflows/Continious%20Integration/badge.svg)
 
+[![Chat on Slack](https://img.shields.io/badge/Chat-on%20Slack-blue)](https://join.slack.com/t/pi4j/shared_invite/zt-1ttqt8wgj-E6t69qaLrNuCMPLiYnBCsg)
```
Comment (taartspi) [inline README.md:8]: when I updated my local repo prior to the latest push it put this change as a new edit, even though this is the code in the remote repo.   Not sure why/how but assume a result of the git commands I used.

**Comment by taartspi** [comment]
@eitch  Robert, I made those edits, thanks for the help.

**Review by eitch — APPROVED** [review]

**Comment by eitch** [comment]
Thanks @taartspi, merged your changes

**Pull request merged**